rest-service/manager_rest/security/authentication.py

```diff
@@ -56,26 +56,55 @@ def authenticate(self, request):
     def _authenticate_password(self, user, auth):
         self.logger.debug('Authenticating username/password')
         username, password = auth.username, auth.password
-        if self.ldap:
-            self.logger.debug('Running LDAP authentication')
-            self.ldap.authenticate_user(username, password)
-            if user:
-                user = self.ldap.update_user(user)
-            else:
-                user = self.ldap.create_user(username)
+
+        # If LDAP is not configured, *or* if the user is the bootstrap admin
+        # we should run a basic HTTP auth
+        if (user and user.is_bootstrap_admin) or not self.ldap:
+            return self._http_auth(user, username, password)
+        # Otherwise, we authenticate through LDAP
         else:
-            self.logger.debug('Running basic HTTP authentication')
-            if not user:
-                raise_unauthorized_user_error(
-                    'Authentication failed for '
-                    '<User username=`{0}`>'.format(username)
-                )
-            if not verify_password(password, user.password):
-                raise_unauthorized_user_error(
-                    'Authentication failed for {0}'.format(user)
-                )
+            return self._ldap_auth(user, username, password)
+
+    def _http_auth(self, user, username, password):
+        """Perform basic user authentication
+        - Check that the password that was passed in the request can be
+          verified against the password stored in the DB
+
+        :param user: The DB user object
+        :param username: The username from the request
+        :param password: The password from the request
+        :return: The DB user object
+        """
+        self.logger.debug('Running basic HTTP authentication')
+        if not user:
+            raise_unauthorized_user_error(
+                'Authentication failed for '
+                '<User username=`{0}`>'.format(username)
```

Comment:
I think that the syntax <User username='''>  gives the impression that it's a print of the User object, but we don't really have User object for that username..